suite-common/connect-popup/src/methodHooks/ethereumSignTransaction.ts

```diff
@@ -0,0 +1,111 @@
+import { Bip43Path, getNetworkByEvmChainId } from '@suite-common/wallet-config';
+import {
+    accountsActions,
+    selectAccounts,
+    selectSelectedDevice,
+    sendFormActions,
+} from '@suite-common/wallet-core';
+import TrezorConnect from '@trezor/connect';
+import type { EthereumSignTransaction } from '@trezor/connect';
+import { getSerializedPath, validatePath } from '@trezor/connect/src/utils/pathUtils';
+
+import { connectPopupActions } from '../connectPopupActions';
+
+import { PreCallHookParams } from './index';
+
+const preCallHook = async <M extends keyof typeof TrezorConnect>({
+    method,
+    payload,
+    getState,
+    dispatch,
+    txSigningPrecomposed,
+}: PreCallHookParams<M>) => {
+    try {
+        if (method === 'ethereumSignTransaction' && txSigningPrecomposed) {
+            const typedPayload = payload as any as EthereumSignTransaction;
+            const device = selectSelectedDevice(getState());
+            const path = getSerializedPath(validatePath(typedPayload.path)) as Bip43Path;
+            const network = getNetworkByEvmChainId(typedPayload.transaction.chainId || 1) || {
+                // Placeholder for chains not supported in Suite
+                networkType: 'ethereum',
+                symbol: 'eth',
+                name: 'Chain ID: ' + typedPayload.transaction.chainId,
+                isHidden: true,
+            };
+            // Try to find matching account
+            let selectedAccount = selectAccounts(getState()).find(
+                account =>
+                    path === account.path && account.symbol === network.symbol && !network.isHidden,
```

Comment:
there is already a selector for almost any combination of filters in accountReducer. also does it make sense to use 
```
&& !network.isHidden,
```
inside find? if network.isHidden, find doesn't need to be called at all?

Reply:
It's kind of a hack-simplification, I'm using isHidden to differentiate unsupported network.